pkg/vm/engine/tae/catalog/block.go

```diff
@@ -72,10 +72,11 @@ func NewBlockEntryWithMeta(
 		segment:       segment,
 		state:         state,
 	}
+	e.MetaBaseEntry.CreateWithTxnAndMeta(txn, metaLoc, deltaLoc)
 	if dataFactory != nil {
 		e.blkData = dataFactory(e)
 	}
-	e.MetaBaseEntry.CreateWithTxnAndMeta(txn, metaLoc, deltaLoc)
+	//e.MetaBaseEntry.CreateWithTxnAndMeta(txn, metaLoc, deltaLoc)
```

Comment:
Is moving this line up intentional?   Add a real comment useful comment and remove this line.

Reply:
Yes , it's intentional.